OpenKeychain/src/main/java/org/sufficientlysecure/keychain/service/BackupNotificationService.java

```diff
@@ -0,0 +1,88 @@
+package org.sufficientlysecure.keychain.service;
+
+import android.app.AlarmManager;
+import android.app.NotificationManager;
+import android.app.PendingIntent;
+import android.app.Service;
+import android.content.Context;
+import android.content.Intent;
+import android.os.IBinder;
+import android.support.annotation.Nullable;
+
+import org.sufficientlysecure.keychain.receiver.BackupNotificationAlarmReceiver;
+import org.sufficientlysecure.keychain.util.Preferences;
+
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * This class is used to setup a backup reminder for every new key created by user
+ */
+public class BackupNotificationService extends Service {
```

Comment:
I'm not so sure about that, firing off an alarmmanager intent seems like a harmless thing for the ui thread, my intuition would be to keep it on there for simplicity. Otherwise, an AsyncTask is a good lightweight option. If you want to go the extra mile here you can of course profile it to get the right answer™.